syntax = "proto3";
import "InventoryMessage.proto";
import "Struct/CommonStruct.proto";
import "Struct/ItemStruct.proto";
package GameMessage;

enum proto_ids 
{
    EMPTY_PROTO = 0;

    // service id
    GAME_SERVICE = 0x0010;

    // command id
    PLAYERONLINE = 0x0001;
    PULL = 0x0002;
}


// sender
message PlayerOnlineRequest
{
    int32 rid = 1;
    int32 eid = 2;
    int32 uid = 3;
}

message PlayerOnlineResponse
{
    int32 result = 1;
    int32 eid = 2;
}

// sender
message PullResponse
{
    int32 result = 1;
    repeated CommonStruct.NetPlayerData players = 2;
    repeated CommonStruct.NetRobotData robots = 3;
    repeated CommonStruct.NetComputerData computers = 4;
    // 推送所有种类物品的基本信息
    repeated ItemStruct.AiItemInfo ai_items_info = 5;
    repeated ItemStruct.EquipItemInfo equip_items_info = 6;
    // 同步背包信息
    repeated InventoryMessage.BackpackInfo backpack_info = 7;
}